waiter/src/waiter/reporter.clj

```diff
@@ -87,15 +100,100 @@
                s/Any s/Any}
               (merge {:pickled? true} config)))
 
+(defn- make-reporting-name
+  "Concatenates elements to form a dotted name for reporting, eliding any null values or empty strings."
+  [prefix & args]
+  (MetricRegistry/name prefix (into-array args)))
+
+(let [df (DecimalFormat. "#.####################")]
+  (defn- format-value
+    "Convert double to string to send to graphite server"
+    [value]
+    (.format df value)))
+
+(defn- report-meter
+  "Report values from a codahale meter"
+  [prefix name ^Metered meter timestamp ^GraphiteSender graphite]
+  (let [send #(.send graphite (make-reporting-name prefix name %1) (format-value (%2 meter)) timestamp)]
```

Comment:
Would it be better to create this function as a `defn`? It seems to be used in both `report-meter` and `report-snapshot`.

Reply:
I meant: `#(.send graphite (make-reporting-name prefix name %1) (format-value (%2 meter)) timestamp)`